internal/expr/parser.go

```diff
@@ -17,20 +19,82 @@ type queryPart interface {
 	part()
 }
 
+type typeSpecifier interface {
+	Name() string
```

Comment:
Definitely, will do that. As for the second part I am not sure because the fields would have nothing to do with each other, we could have an enum or a flag to create a poor man's union type though. I will think about it for the merge and then we can discuss.

Reply:
Agreed, I think an interface is a good option, though something worth considering is that it could be an interface over the `memberName` rather than whole `valueAccessor` since both slices and regular dot expressions have a `typeName`